content/en/docs/concepts/storage/storage-classes.md

```diff
@@ -15,59 +15,80 @@ This document describes the concept of a StorageClass in Kubernetes. Familiarity
+As an administrator, you can specify a default StorageClass that applies to any PVCs that
+don't request any particular class to bind to; see the
+[PersistentVolumeClaim concept](/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims)
+for more details.
 
-```yaml
-apiVersion: storage.k8s.io/v1
-kind: StorageClass
-metadata:
-  name: standard
-provisioner: kubernetes.io/aws-ebs
-parameters:
-  type: gp2
-reclaimPolicy: Retain
-allowVolumeExpansion: true
-mountOptions:
-  - debug
-volumeBindingMode: Immediate
-```
+Here's an example of a StorageClass:
 
-### Default StorageClass
+{{% code file="storage/storageclass-low-latency.yaml" %}}
 
-When a PVC does not specify a `storageClassName`, the default StorageClass is
-used. The cluster can only have one default StorageClass. If more than one
-default StorageClass is accidentally set, the newest default is used when the
-PVC is dynamically provisioned.
+## Default StorageClass
 
+You can mark a StorageClass as the default for your cluster.
 For instructions on setting the default StorageClass, see
 [Change the default StorageClass](/docs/tasks/administer-cluster/change-default-storage-class/).
-Note that certain cloud providers may already define a default StorageClass.
 
-### Provisioner
+When a PVC does not specify a `storageClassName`, the default StorageClass is
+used.
+
+If you set the
+[`storageclass.kubernetes.io/is-default-class`](/docs/reference/labels-annotations-taints/#ingressclass-kubernetes-io-is-default-class)
+annotation to true on more than one StorageClass in your cluster, and you then
+create a PersistentVolumeClaim with no `storageClassName` set, Kubernetes
+picks which of the “default” StorageClasses to use based on which was created
+more recently.
+
+{{< note >}}
+You should try to only have one StorageClass in your cluster that is
+marked as the default. The reason that Kubernetes allows you to have
+two (or more) default StorageClasses is to allow for seamless migration.
+{{< /note >}}
+
+You can create a PersistentVolumeClaim without specifying a `storageClassName`
+for the new PVC, and you can do so even when no default StorageClass exists
+in your cluster. In this case, the new PVC creates as you defined it, and the
+`storageClassName` of that PVC remains unset until default becomes available.
+
+You can have a cluster without any default StorageClass. If you don't mark any
+StorageClass as default (and your cloud provider, etc has not set one for you),
+then Kubernetes cannot apply that defaulting for PersistentVolumeClaims that need
+it.
+
+If or when a default StorageClass becomes available, the control plane identifies any
+existing PVCs without `storageClassName`. For the PVCs that either have an empty
+value for `storageClassName` or do not have this key, the control plane then
+updates those PVCs to set `storageClassName` to match the new default StorageClass.
+If you have an existing PVC where the `storageClassName` is `""`, and you configure
```

Comment:
This statement is inconsistent with the previous one.
If `storageClassName: ""` is specified, a new default StorageClass is then added, what will happen?

Reply:
(this text is copied from https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retroactive-default-storageclass-assignment; if that text is wrong, we should fix both occurrences).

Reply:
To me, who knows how default storage classes work, the PR text looks correct.

However, it would be great to distinguish between "empty value" (the filed is missing or `nil`) and empty string `""` more explicitly. How it's done in other API fields where `nil` and `""` means something different?